### DISCUSSION TITLE
How to pass  input value by name to api url of dropdown?

### DISCUSSION THREAD

**Discussion by viratsoft** [fomantic/Fomantic-UI#2100] [Modules (Modal, Dropdown, Popup, Calendar, Toast, ...)] [answered]
how can i send input value to dropdown api url from another input value by name?

when i try like bellow but not send current value of filter_id input to api url of dropdown.

```
.dropdown({
            apiSettings: {
                url: '/get_category.php?type='+ $('input[name="filter_id"]').val(),
                cache: false
            },
```

thank you

**Comment by lubber-de** [comment, marked as answer]
Your approach adds the input field value only once when the dropdown is initiated.
Instead use a placeholder for the type parameter and change that before the api is sending to the backend.

```javascript
.dropdown({
	apiSettings: {
		url: '/get_category.php?type={type}',
		beforeSend: function(settings) {
		  settings.urlData = {
			type: $('input[name="filter_id"]').val()
		  };
		  return settings;
		},
		cache: false
	}
});
```

https://fomantic-ui.com/behaviors/api.html#4-settings-returned-from-beforesend

**Reply by viratsoft** [reply]
solved,
thank you so much...
